db.go

```diff
@@ -804,32 +802,48 @@ func (d *DB) walPreallocateSize() int {
 
 func (d *DB) makeRoomForWrite(b *Batch) error {
 	force := b == nil || b.flushable != nil
+	stalled := false
 	for {
```

Comment:
tbh I do not fully grok this loop. Is calling this function with `b == nil` supposed to always create a new WAL/memtable?